v1-2/learn/how-to-deploy-and-run-ballerina-programs.md

```diff
@@ -508,86 +532,110 @@ Add an `/etc/hosts` entry to match hostname.
 ```bash
 127.0.0.1 ballerina.guides.io
 ```
+
+Setup the NGINX Ingress Controller.
+
+```bash
+kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/nginx-0.27.0/deploy/static/mandatory.yaml
+kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/nginx-0.27.0/deploy/static/provider/cloud-generic.yaml
+kubectl patch deployments -n ingress-nginx nginx-ingress-controller -p '{"spec":{"template":{"spec":{"containers":[{"name":"nginx-ingress-controller","args":["\/nginx-ingress-controller","--configmap=$(POD_NAMESPACE)\/nginx-configuration","--tcp-services-configmap=$(POD_NAMESPACE)\/tcp-services","--udp-services-configmap=$(POD_NAMESPACE)\/udp-services","--publish-service=$(POD_NAMESPACE)\/ingress-nginx","--annotations-prefix=nginx.ingress.kubernetes.io","--annotations-prefix=nginx.ingress.kubernetes.io","--enable-ssl-passthrough"]}]}}}}'
+```
+
 Access the service:
 
 ```bash
-$ curl -v -X POST -d '{"name":"Alice", "age":20,"ssn":123456789,"employeeId":1}' \
-"http://ballerina.guides.io/records/employee" -H "Content-Type:application/json"
+$ curl -kv https://ballerina.guides.io/users/jane 
+{"name":"Jane Doe", "email":"jane@ballerina.com"}
 ```   
     
 ##### Supported Kubernetes Annotations
 
 **@kubernetes:Deployment{}**
-- Supported with Ballerina services or endpoints.
+- Supported with Ballerina services, listeners and functions.
 
 |**Annotation Name**|**Description**|**Default value**|
 |--|--|--|
-|name|Name of the deployment|\<outputfilename\>-deployment|
-|labels|Labels for deployment|"app: \<outputfilename\>"|
+|name|Name of the deployment|<BALLERINA_FILE_NAME>-deployment or <BALLERINA_MODULE_NAME>-deployment|
+|labels|Labels for deployment|{ app: <OUTPUT_FILE_NAME> }|
+|annotations|Annotations for deployment|{}|
+|dockerHost|Docker host IP and docker PORT.(e.g "tcp://192.168.99.100:2376")|DOCKER_HOST environment variable. If DOCKER_HOST is unavailable, uses "unix:///var/run/docker.sock" for Unix or uses "npipe:////./pipe/docker_engine" for Windows 10 or uses "localhost:2375"|
+|dockerCertPath|Docker cert path|DOCKER_CERT_PATH environment variable|
+|registry|Docker registry url|null|
+|username|Username for the docker registry|null|
+|password|Password for the docker registry|null|
+|baseImage|Base image to create the docker image|ballerina/jre8:v1|
+|image|Docker image with tag|<OUTPUT_FILE_NAME>:latest. If field `registry` is set then it will be prepended to the docker image name as <registry>/<OUTPUT_FILE_NAME>:latest|

```

Comment:
```suggestion
|image|Docker image with tag|<OUTPUT_FILE_NAME>:latest. If field `registry` is set ,then it will be prepended to the Docker image name as <registry>/<OUTPUT_FILE_NAME>:latest|

```